param.go

```diff
@@ -231,10 +228,13 @@ func (ps paramSingle) Build(c *Container) (reflect.Value, error) {
 			return reflect.Zero(ps.Type), nil
 		}
 
-		return _noValue, errParamSingleFailed{Key: k, Reason: err}
+		return _noValue, errParamSingleFailed{Key: key{t: ps.Type, name: ps.Name}, Reason: err}
```

Comment:
nit: consider using the same static construction in the error case above for consistency:
```
key{name: ps.Name, t: ps.Type}
```